### DISCUSSION TITLE
My BirdNET-Pi website down frequently

### DISCUSSION THREAD

**Discussion by ekimyar** [mcguirepr89/BirdNET-Pi#862] [Q&A] [answered]
I installed birdnet-pi on a Libre Computers Renegade with 4gb of RAM on a 64gb SanDisk Extreme micro SD Card, running the Libre-tweaked Raspberry Pi Lite OS. It was running pretty well with occasional website outages (i.e. the birdnetpi.local website would not load), although I was almost always able to SSH into the Renegade. In the last couple days though, it is far more often down than up. I can still SSH, but restarting services (using the restart_services.sh command in the scrips folder) or just rebooting doesn't really seem to help. What would you suggest I look at to get this working better?

**Comment by ehpersonal38** [comment]
Could you run the following command on your system and paste the output here so I can take a look?

`bash ~/BirdNET-Pi/scripts/print_diagnostic_info.sh`

**Reply by ekimyar** [reply]
Thanks very much for the help. It apparently sensed that I asked for help, because it's been up far more often than down this last 24 hours. Diagnostic info pasted below!


========== caddy status ==========
● caddy.service - Caddy
     Loaded: loaded (/lib/systemd/system/caddy.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/caddy.service.d
             └─override.conf
     Active: active (running) since Mon 2023-04-17 20:54:14 PDT; 1 day 18h ago
       Docs: https://caddyserver.com/docs/
   Main PID: 1081 (caddy)
      Tasks: 10 (limit: 4291)
     Memory: 56.2M
        CPU: 56.837s
     CGroup: /system.slice/caddy.service
             └─1081 /usr/bin/caddy run --environ --config /etc/caddy/Caddyfile

Apr 17 20:54:13 birdnetpi caddy[1081]: {"level":"warn","ts":1681790053.9868312,"logger":"http","msg":"server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server","server_name":"srv0","http_port":80}
Apr 17 20:54:13 birdnetpi caddy[1081]: {"level":"info","ts":1681790053.9958858,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x400042d420"}
Apr 17 20:54:14 birdnetpi caddy[1081]: {"level":"info","ts":1681790054.0026083,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/var/lib/caddy/.local/share/caddy"}
Apr 17 20:54:14 birdnetpi caddy[1081]: {"level":"info","ts":1681790054.0044615,"logger":"tls","msg":"finished cleaning storage units"}
Apr 17 20:54:14 birdnetpi caddy[1081]: {"level":"info","ts":1681790054.006446,"logger":"http.log","msg":"server running","name":"srv0","protocols":["h1","h2","h3"]}
Apr 17 20:54:14 birdnetpi caddy[1081]: {"level":"info","ts":1681790054.0173013,"msg":"autosaved config (load with --resume flag)","file":"/var/lib/caddy/.config/caddy/autosave.json"}
Apr 17 20:54:14 birdnetpi systemd[1]: Started Caddy.
Apr 17 20:54:14 birdnetpi caddy[1081]: {"level":"info","ts":1681790054.040647,"msg":"serving initial configuration"}
Apr 18 20:54:13 birdnetpi caddy[1081]: {"level":"info","ts":1681876453.3892694,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/var/lib/caddy/.local/share/caddy"}
Apr 18 20:54:13 birdnetpi caddy[1081]: {"level":"info","ts":1681876453.3894255,"logger":"tls","msg":"finished cleaning storage units"}

========== birdnet_analysis status ==========
● birdnet_analysis.service - BirdNET Analysis
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/birdnet_analysis.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-04-19 15:41:56 PDT; 12s ago
   Main PID: 2463892 (bash)
      Tasks: 2 (limit: 4291)
     Memory: 6.2M
        CPU: 1.006s
     CGroup: /system.slice/birdnet_analysis.service
             ├─2463892 bash /usr/local/bin/birdnet_analysis.sh
             └─2464110 /home/birdpi/BirdNET-Pi/birdnet/bin/python3 /home/birdpi/BirdNET-Pi/scripts/analyze.py --i /home/birdpi/BirdSongs/April-2023/19-Wednesday/2023-04-19-birdnet-15:41:51.wav --o /home/birdpi/BirdSongs/April-2023/19-Wednesday/2023-04-19-birdnet-15:41:51.wav.csv --lat 32.6889 --lon -117.2408 --week 15 --overlap 0.0 --sensitivity 1.25 --min_conf 0.7

Apr 19 15:42:10 birdnetpi birdnet_analysis.sh[2464110]: 9.0;12.0-("Leiothlypis luciae_Lucy's Warbler", 0.098923974)
Apr 19 15:42:10 birdnetpi birdnet_analysis.sh[2464110]: 12.0;15.0-('Haemorhous mexicanus_House Finch', 0.16840015)
Apr 19 15:42:10 birdnetpi systemd[1]: birdnet_analysis.service: Succeeded.
Apr 19 15:42:10 birdnetpi systemd[1]: birdnet_analysis.service: Consumed 1.062s CPU time.
Apr 19 15:42:12 birdnetpi systemd[1]: birdnet_analysis.service: Scheduled restart job, restart counter is at 10274.
Apr 19 15:42:12 birdnetpi systemd[1]: Stopped BirdNET Analysis.
Apr 19 15:42:12 birdnetpi systemd[1]: birdnet_analysis.service: Consumed 1.062s CPU time.
Apr 19 15:42:12 birdnetpi systemd[1]: Started BirdNET Analysis.
Apr 19 15:42:13 birdnetpi birdnet_analysis.sh[2464186]: Files loaded
Apr 19 15:42:14 birdnetpi birdnet_analysis.sh[2464186]: RECORDING_LENGTH set to 15

========== birdnet_log status ==========
● birdnet_log.service - BirdNET Analysis Log
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/birdnet_log.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2023-04-18 07:01:58 PDT; 1 day 8h ago
   Main PID: 586787 (gotty)
      Tasks: 6 (limit: 4291)
     Memory: 12.6M
        CPU: 131ms
     CGroup: /system.slice/birdnet_log.service
             └─586787 /usr/local/bin/gotty --address localhost -p 8080 -P log --title-format BirdNET-Pi Log birdnet_log.sh

Apr 18 07:01:58 birdnetpi systemd[1]: Started BirdNET Analysis Log.
Apr 18 07:01:58 birdnetpi gotty[586787]: 2023/04/18 07:01:58 Loading config file at: /home/birdpi/.gotty
Apr 18 07:01:58 birdnetpi gotty[586787]: 2023/04/18 07:01:58 GoTTY is starting with command: birdnet_log.sh
Apr 18 07:01:58 birdnetpi gotty[586787]: 2023/04/18 07:01:58 HTTP server is listening at: http://127.0.0.1:8080/log/

========== birdnet_recording status ==========
● birdnet_recording.service - BirdNET Recording
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/birdnet_recording.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-04-19 13:19:21 PDT; 2h 22min ago
   Main PID: 2325504 (bash)
      Tasks: 3 (limit: 4291)
     Memory: 141.7M
        CPU: 1min 42.593s
     CGroup: /system.slice/birdnet_recording.service
             ├─2325504 bash /usr/local/bin/birdnet_recording.sh
             └─2325530 arecord -f S16_LE -c2 -r48000 -t wav --max-file-time 15 -D default --use-strftime /home/birdpi/BirdSongs/%B-%Y/%d-%A/%F-birdnet-%H:%M:%S.wav

Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325504]: + '[' -z 15 ']'
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325504]: + '[' '!' -z ']'
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325504]: + pulseaudio --check
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325504]: + pgrep arecord
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325504]: + grep 5050 /dev/fd/63
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325521]: ++ netstat -tulpn
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325522]: tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      586823/python3
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325504]: + '[' -z default ']'
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325504]: + arecord -f S16_LE -c2 -r48000 -t wav --max-file-time 15 -D default --use-strftime /home/birdpi/BirdSongs/%B-%Y/%d-%A/%F-birdnet-%H:%M:%S.wav
Apr 19 13:19:21 birdnetpi birdnet_recording.sh[2325530]: Recording WAVE '/home/birdpi/BirdSongs/%B-%Y/%d-%A/%F-birdnet-%H:%M:%S.wav' : Signed 16 bit Little Endian, Rate 48000 Hz, Stereo

========== birdnet_server status ==========
● birdnet_server.service - BirdNET Analysis Server
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/birdnet_server.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2023-04-18 07:02:00 PDT; 1 day 8h ago
   Main PID: 586823 (python3)
      Tasks: 9 (limit: 4291)
     Memory: 229.1M
        CPU: 9h 20min 10.064s
     CGroup: /system.slice/birdnet_server.service
             └─586823 /home/birdpi/BirdNET-Pi/birdnet/bin/python3 /usr/local/bin/server.py

Apr 19 15:41:39 birdnetpi python3[586823]: READING AUDIO DATA... DONE! READ 5 CHUNKS.
Apr 19 15:41:39 birdnetpi python3[586823]: Date-time: 2023-04-19 15:41:21
Apr 19 15:41:52 birdnetpi python3[586823]: ANALYZING AUDIO... DONE! Time 2.5 SECONDS
Apr 19 15:41:52 birdnetpi python3[586823]: WRITING RESULTS TO /home/birdpi/BirdSongs/April-2023/19-Wednesday/2023-04-19-birdnet-15:41:21.wav.csv ... DONE! WROTE 0 RESULTS.
Apr 19 15:41:52 birdnetpi python3[586823]: READING AUDIO DATA... DONE! READ 5 CHUNKS.
Apr 19 15:41:52 birdnetpi python3[586823]: Date-time: 2023-04-19 15:41:36
Apr 19 15:42:07 birdnetpi python3[586823]: ANALYZING AUDIO... DONE! Time 2.4 SECONDS
Apr 19 15:42:07 birdnetpi python3[586823]: WRITING RESULTS TO /home/birdpi/BirdSongs/April-2023/19-Wednesday/2023-04-19-birdnet-15:41:36.wav.csv ... DONE! WROTE 0 RESULTS.
Apr 19 15:42:07 birdnetpi python3[586823]: READING AUDIO DATA... DONE! READ 5 CHUNKS.
Apr 19 15:42:07 birdnetpi python3[586823]: Date-time: 2023-04-19 15:41:51

========== birdnet_stats status ==========
● birdnet_stats.service - BirdNET Stats
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/birdnet_stats.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2023-04-18 07:02:00 PDT; 1 day 8h ago
   Main PID: 586819 (streamlit)
      Tasks: 5 (limit: 4291)
     Memory: 106.0M
        CPU: 3min 53.264s
     CGroup: /system.slice/birdnet_stats.service
             └─586819 /home/birdpi/BirdNET-Pi/birdnet/bin/python3 /home/birdpi/BirdNET-Pi/birdnet/bin/streamlit run /home/birdpi/BirdNET-Pi/scripts/plotly_streamlit.py --browser.gatherUsageStats false --server.address localhost --server.baseUrlPath /stats

Apr 18 07:02:00 birdnetpi systemd[1]: Started BirdNET Stats.
Apr 18 07:02:13 birdnetpi streamlit[586819]:   You can now view your Streamlit app in your browser.
Apr 18 07:02:13 birdnetpi streamlit[586819]:   URL: http://localhost:8501/stats

========== chart_viewer status ==========
● chart_viewer.service - BirdNET-Pi Chart Viewer Service
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/chart_viewer.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-04-19 15:42:07 PDT; 12s ago
   Main PID: 2464112 (python3)
      Tasks: 8 (limit: 4291)
     Memory: 120.9M
        CPU: 12.973s
     CGroup: /system.slice/chart_viewer.service
             └─2464112 /home/birdpi/BirdNET-Pi/birdnet/bin/python3 /usr/local/bin/daily_plot.py

Apr 19 15:42:07 birdnetpi systemd[1]: Started BirdNET-Pi Chart Viewer Service.

========== extraction status ==========
● extraction.service - BirdNET BirdSound Extraction
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/extraction.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2023-04-18 07:01:58 PDT; 1 day 8h ago
   Main PID: 586769 (bash)
      Tasks: 2 (limit: 4291)
     Memory: 573.1M
        CPU: 3h 38min 2.029s
     CGroup: /system.slice/extraction.service
             ├─ 586769 bash -c while true;do extract_new_birdsounds.sh;sleep 3;done
             └─2464332 sleep 3

Apr 19 15:40:02 birdnetpi env[2462113]: Line = 2023-04-19-birdnet-15:39:36.wav.csv
Apr 19 15:40:15 birdnetpi env[2462319]: Line = 2023-04-19-birdnet-15:39:51.wav.csv
Apr 19 15:40:28 birdnetpi env[2462532]: Line = 2023-04-19-birdnet-15:40:06.wav.csv
Apr 19 15:40:45 birdnetpi env[2462795]: Line = 2023-04-19-birdnet-15:40:21.wav.csv
Apr 19 15:41:02 birdnetpi env[2463030]: Line = 2023-04-19-birdnet-15:40:36.wav.csv
Apr 19 15:41:15 birdnetpi env[2463247]: Line = 2023-04-19-birdnet-15:40:51.wav.csv
Apr 19 15:41:32 birdnetpi env[2463495]: Line = 2023-04-19-birdnet-15:41:06.wav.csv
Apr 19 15:41:45 birdnetpi env[2463722]: Line = 2023-04-19-birdnet-15:41:21.wav.csv
Apr 19 15:41:58 birdnetpi env[2463932]: Line = 2023-04-19-birdnet-15:41:36.wav.csv
Apr 19 15:42:15 birdnetpi env[2464255]: Line = 2023-04-19-birdnet-15:41:51.wav.csv

========== web_terminal status ==========
● web_terminal.service - BirdNET-Pi Web Terminal
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/web_terminal.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2023-04-17 20:54:01 PDT; 1 day 18h ago
   Main PID: 698 (gotty)
      Tasks: 7 (limit: 4291)
     Memory: 6.6M
        CPU: 150ms
     CGroup: /system.slice/web_terminal.service
             └─698 /usr/local/bin/gotty --address localhost -w -p 8888 -P terminal --title-format BirdNET-Pi Terminal login

Apr 17 20:54:02 birdnetpi gotty[698]: 2023/04/17 20:54:01 GoTTY is starting with command: login
Apr 17 20:54:02 birdnetpi gotty[698]: 2023/04/17 20:54:01 Permitting clients to write input to the PTY.
Apr 17 20:54:02 birdnetpi gotty[698]: 2023/04/17 20:54:01 HTTP server is listening at: http://127.0.0.1:8888/terminal/
Apr 17 20:54:01 birdnetpi systemd[1]: Started BirdNET-Pi Web Terminal.

========== spectrogram_viewer status ==========
● spectrogram_viewer.service - BirdNET-Pi Spectrogram Viewer
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/spectrogram_viewer.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-04-19 15:42:23 PDT; 236ms ago
   Main PID: 2464400 (bash)
      Tasks: 2 (limit: 4291)
     Memory: 600.0K
        CPU: 22ms
     CGroup: /system.slice/spectrogram_viewer.service
             ├─2464400 bash /usr/local/bin/spectrogram.sh
             └─2464410 bash /usr/local/bin/spectrogram.sh

Apr 19 15:42:23 birdnetpi systemd[1]: Stopped BirdNET-Pi Spectrogram Viewer.
Apr 19 15:42:23 birdnetpi systemd[1]: Started BirdNET-Pi Spectrogram Viewer.
Apr 19 15:42:26 birdnetpi systemd[1]: spectrogram_viewer.service: Succeeded.

========== livestream status ==========
● livestream.service - BirdNET-Pi Live Stream
     Loaded: loaded (/home/birdpi/BirdNET-Pi/templates/livestream.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2023-04-18 07:02:00 PDT; 1 day 8h ago
   Main PID: 586817 (bash)
      Tasks: 7 (limit: 4291)
     Memory: 10.4M
        CPU: 8h 28min 54.914s
     CGroup: /system.slice/livestream.service
             ├─586817 bash /usr/local/bin/livestream.sh
             └─586818 ffmpeg -nostdin -loglevel 32 -ac 2 -f alsa -i default -acodec libmp3lame -b:a 320k -ac 2 -content_type audio/mpeg -f mp3 icecast://source:birdnetpi@localhost:8000/stream -re

Apr 19 14:49:46 birdnetpi livestream.sh[586818]: [48.0K blob data]
Apr 19 14:55:34 birdnetpi livestream.sh[586818]: [48.0K blob data]
Apr 19 15:01:24 birdnetpi livestream.sh[586818]: [48.0K blob data]
Apr 19 15:07:13 birdnetpi livestream.sh[586818]: [48.0K blob data]
Apr 19 15:13:01 birdnetpi livestream.sh[586818]: [47.9K blob data]
Apr 19 15:18:51 birdnetpi livestream.sh[586818]: [48.0K blob data]
Apr 19 15:24:40 birdnetpi livestream.sh[586818]: [48.0K blob data]
Apr 19 15:30:29 birdnetpi livestream.sh[586818]: [48.0K blob data]
Apr 19 15:36:17 birdnetpi livestream.sh[586818]: [47.9K blob data]
Apr 19 15:42:07 birdnetpi livestream.sh[586818]: [48.0K blob data]

========== Mounted File Systems ==========
Filesystem      Size  Used Avail Use% Mounted on
udev            1.8G     0  1.8G   0% /dev
tmpfs           398M  1.3M  397M   1% /run
/dev/mmcblk1p2   60G   15G   45G  25% /
tmpfs           2.0G     0  2.0G   0% /dev/shm
tmpfs           5.0M  4.0K  5.0M   1% /run/lock
/dev/mmcblk1p1  255M   39M  217M  16% /boot/efi
tmpfs           398M  4.0K  398M   1% /run/user/1000

========== Memory Usage ==========
               total        used        free      shared  buff/cache   available
Mem:           3.9Gi       679Mi        32Mi       4.0Mi       3.2Gi       3.2Gi
Swap:           99Mi       3.0Mi        96Mi

========== Load Averages ==========
 15:42:27 up 1 day, 18:48,  2 users,  load average: 3.99, 2.70, 2.33

========== CPU Info ==========
processor	: 0
BogoMIPS	: 48.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 1
BogoMIPS	: 48.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 2
BogoMIPS	: 48.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4

processor	: 3
BogoMIPS	: 48.00
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd03
CPU revision	: 4


========== System Temperature ==========
CPU Temperature: 65.00°C / 149.00°F

========== Extra Info ==========
........................................IPs.....................................
LAN IP: 192.168.0.99
Public IP: 68.6.184.37
/usr/local/bin/extra_info.sh: command substitution: line 8: syntax error near unexpected token `!'
/usr/local/bin/extra_info.sh: command substitution: line 8: `dpkg -l | ! grep -q "vcgencmd")"'
.....................................Caddyfile..................................
http://  {
  root * /home/birdpi/BirdSongs/Extracted
  file_server browse
  handle /By_Date/* {
    file_server browse
  }
  handle /Charts/* {
    file_server browse
  }
  reverse_proxy /stream localhost:8000
  php_fastcgi unix//run/php/php7.4-fpm.sock
  reverse_proxy /log* localhost:8080
  reverse_proxy /stats* localhost:8501
  reverse_proxy /terminal* localhost:8888
}
.................................... Crontab....................................

SHELL=/bin/sh
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin

17 *	* * *	root    cd / && run-parts --report /etc/cron.hourly
25 6	* * *	root	test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.daily )
47 6	* * 7	root	test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.weekly )
52 6	1 * *	root	test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.monthly )
*/5 * * * * birdpi /usr/local/bin/disk_check.sh >/dev/null 2>&1
*/3 * * * * birdpi /usr/local/bin/cleanup.sh >/dev/null 2>&1
@reboot birdpi /usr/local/bin/cleanup.sh >/dev/null 2>&1
0 1 * * 1 birdpi /usr/local/bin/weekly_report.sh >/dev/null 2>&1

========== Connected Microphone Devices ==========
**** List of CAPTURE Hardware Devices ****
card 0: Analog [Analog], device 0: ff010000.i2s-rk3328-hifi rk3328-hifi-0 [ff010000.i2s-rk3328-hifi rk3328-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: HIDMediak [UAC 1.0 Microphone & HID-Mediak], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 2: HDMI [HDMI], device 0: ff000000.i2s-i2s-hifi i2s-hifi-0 [ff000000.i2s-i2s-hifi i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

null
    Discard all samples (playback) or generate zero samples (capture)
default
    Playback/recording through the PulseAudio sound server
lavrate
    Rate Converter Plugin Using Libav/FFmpeg Library
samplerate
    Rate Converter Plugin Using Samplerate Library
speexrate
    Rate Converter Plugin Using Speex Resampler
jack
    JACK Audio Connection Kit
oss
    Open Sound System
pulse
    PulseAudio Sound Server
upmix
    Plugin for channel upmix (4,6,8)
vdownmix
    Plugin for channel downmix (stereo) with a simple spacialization
hw:CARD=Analog,DEV=0
    Analog, ff010000.i2s-rk3328-hifi rk3328-hifi-0
    Direct hardware device without any conversions
plughw:CARD=Analog,DEV=0
    Analog, ff010000.i2s-rk3328-hifi rk3328-hifi-0
    Hardware device with all software conversions
sysdefault:CARD=Analog
    Analog, ff010000.i2s-rk3328-hifi rk3328-hifi-0
    Default Audio Device
dsnoop:CARD=Analog,DEV=0
    Analog, ff010000.i2s-rk3328-hifi rk3328-hifi-0
    Direct sample snooping device
usbstream:CARD=Analog
    Analog
    USB Stream Output
hw:CARD=HIDMediak,DEV=0
    UAC 1.0 Microphone & HID-Mediak, USB Audio
    Direct hardware device without any conversions
plughw:CARD=HIDMediak,DEV=0
    UAC 1.0 Microphone & HID-Mediak, USB Audio
    Hardware device with all software conversions
sysdefault:CARD=HIDMediak
    UAC 1.0 Microphone & HID-Mediak, USB Audio
    Default Audio Device
front:CARD=HIDMediak,DEV=0
    UAC 1.0 Microphone & HID-Mediak, USB Audio
    Front output / input
dsnoop:CARD=HIDMediak,DEV=0
    UAC 1.0 Microphone & HID-Mediak, USB Audio
    Direct sample snooping device
usbstream:CARD=HIDMediak
    UAC 1.0 Microphone & HID-Mediak
    USB Stream Output
hw:CARD=HDMI,DEV=0
    HDMI, ff000000.i2s-i2s-hifi i2s-hifi-0
    Direct hardware device without any conversions
plughw:CARD=HDMI,DEV=0
    HDMI, ff000000.i2s-i2s-hifi i2s-hifi-0
    Hardware device with all software conversions
sysdefault:CARD=HDMI
    HDMI, ff000000.i2s-i2s-hifi i2s-hifi-0
    Default Audio Device
dsnoop:CARD=HDMI,DEV=0
    HDMI, ff000000.i2s-i2s-hifi i2s-hifi-0
    Direct sample snooping device
usbstream:CARD=HDMI
    HDMI
    USB Stream Output

========= Date and Time ==========
Wed 19 Apr 15:42:33 PDT 2023
===========================================

**Reply by ehpersonal38** [reply, marked as answer]
Thank you. Everything looks good there. I'd be interested in seeing that same output when your issues arrise. Then we can compare the outputs and see what is different about them, and hopefully learn more about the problem.

**Reply by ekimyar** [reply]
Thanks very much. When it gets weird again I'll be sure to do that.

**Comment by skyjoe66** [comment]
I was having a similar issue with the web site being down but the pi working fine. I ran the diagnostic script above suggested by @ehpersonal38  (thank you!) and had chatgpt take a look to find errors.  My caddy service was failing. "Error: ambiguous config file format; please specify adapter (use --adapter)"

GPT recommended editing /etc/systemd/system/caddy.service.d/override.conf and adding the following two lines: 
[Service]
ExecStart=
ExecStart=/usr/bin/caddy run --environ --config /etc/caddy/Caddyfile --adapter caddyfile

then 
sudo systemctl daemon-reload
sudo systemctl restart caddy.service

It appears to be working now.
